Move docs readme to separate file (#5710)
treeverse/lakeFS: docs/README.md

# lakeFS Documentation

Any contribution to the docs, whether it is in conjunction with a code contribution or as a standalone, is appreciated.

Please see [the contributing guide](contributing.md) for details on contributing to lakeFS in general. 

> 📝  Notice! lakeFS documentation is written using Markdown. make sure to familiarize yourself with the [Markdown Guide](https://www.markdownguide.org/basic-syntax/#heading-best-practices).

Reviewer: Nit:
```suggestion
📝  Notice! lakeFS documentation is written using Markdown.  Make sure to familiarize yourself with the [Markdown Guide](https://www.markdownguide.org/basic-syntax/#heading-best-practices).
```

Customizing the lakeFS docs site should follow the following guidelines: [Just The Docs Customization](https://just-the-docs.github.io/just-the-docs/docs/customization/) and style-guide.

### lakeFS Style Guide:
* Don't use unnecessary tech jargon or vague/wordy constructions - keep it friendly, not condescending.

> * Be inclusive and welcoming - use gender-neutral words and pronouns when talking about abstract people like developers).

Reviewer: Nit:
```suggestion
* Be inclusive and welcoming - use gender-neutral words and pronouns when talking about abstract people like users and developers.
```

* Replace complex expressions with simpler ones.
* Keep it short - 25-30 words max per sentence.  Otherwise, your readers might get lost on the way. 
* Use active voice instead of passive. For example: This feature can be used to do task X. vs. You can use this feature to do task X. The second one reads much better, right?
* You can explain things better by including examples. Show, not tell. Use illustrations, images, gifs, code snippets, etc.

> * Establish a visual hierarchy to help people quickly find the information they need. Use text formatting to create levels of title and subtitle (such as h1 to h6 headings in HTML).

Reviewer: *Suggestion:*
```suggestion
* Establish a visual hierarchy to help people quickly find the information they need. Use text formatting to create levels of title and subtitle (such as `#` to `######` markdown headings).  The title of every page should use the topmost heading `#`; all other headings on the page should use lower headers `##` to `######`.
```

### Test your changes locally

> To render the documentation locally and preview changes you can run the Jeykll server under Docker. 

Reviewer: It should also be possible to `make docs-serve`, assuming you have the relevant tooling.  Do we want to mention that option?

Author: Good point. I'll mention it.

1. Launch the Docker container:

   ```sh
   docker run --rm \
              --name lakefs_docs \
              -e TZ="Etc/UTC" \
              --publish 4000:4000 --publish 35729:35729 \
              --volume="$PWD/docs:/srv/jekyll:Z" \
              --volume="$PWD/docs/.jekyll-bundle-cache:/usr/local/bundle:Z" \
              --interactive --tty \
              jekyll/jekyll:3.8 \
              jekyll serve --livereload

>    ```

Reviewer: Consider making this a Makefile target.

Author: as well as, or instead of, the existing `docs-serve` ?

Reviewer: Probably in addition: Makefiles that just run Docker images are a bit of a nuisance for developers.  Perhaps `docs-serve-docker` would be a better name??...?

Author: Done
